packages/docusaurus-theme-classic/src/index.ts

```diff
@@ -156,29 +153,21 @@ export default function docusaurusThemeClassic(
       };
     },
 
-    configurePostCss(postCssOptions) {
+    configurePostCss(postCssOptions: {plugins: AcceptedPlugin[]}) {
       if (direction === 'rtl') {
-        postCssOptions.plugins.push(
-          postcss.plugin('RtlCssPlugin', () => {
-            const resolvedInfimaFile = require.resolve(
-              getInfimaCSSFile(direction),
-            );
-            function isInfimaCSSFile(file?: string) {
-              return file === resolvedInfimaFile;
+        const resolvedInfimaFile = require.resolve(getInfimaCSSFile(direction));
```

Comment:
It was better to create `resolvedInfimaFile` variable in `prepare` method (see https://evilmartians.com/chronicles/postcss-8-plugin-migration):


```ts
 const plugin: PostCssPlugin = {
  postcssPlugin: 'RtlCssPlugin',
  prepare() {
    const resolvedInfimaFile = require.resolve(
      getInfimaCSSFile(direction),
    );

    return {
      Once(root) {
        const file = root?.source?.input.file;

        // Skip Infima as we are using the its RTL version.
        if (file === resolvedInfimaFile) {
          return;
        }

        rtlcss.process(root);
      },
    };
  },
};
```

Reply:
I don't think this is needed. 
Here we don't need to use shared state betweens postcss listeners on a per-file basis, but we just need to detect if the current file is the infima file. The resolve function will always return the same result so we'd rather just execute it once instead of executing for each css file.